pkg/userdata/helper/kubelet.go

```diff
@@ -58,6 +59,9 @@ const (
 {{- if .PauseImage }}
 --pod-infra-container-image={{ .PauseImage }} \
 {{- end }}
+{{- if .InitialTaints }}
+--register-with-taints={{- range $i, $t := .InitialTaints }}{{ $t.Key }}={{ $t.Value }}:{{ $t.Effect }}{{ if (lt $i (sub (len $.InitialTaints) 1)) }},{{ end }}{{ end }} \
```

Comment:
Probably this logic can be moved to `KubeletFlags()` and be pre-computed before template render. Would be much easier to reason about it.